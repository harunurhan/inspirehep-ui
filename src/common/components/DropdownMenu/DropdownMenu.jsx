import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Menu, Dropdown, Icon } from 'antd';

class DropdownMenu extends Component {
  static renderAnchorMenuItem(item) {
    if (item.href) {
      return (
        <a target={item.target} href={item.href}>
          {item.display}
        </a>
      );
    }
    return (
      // eslint-disable-next-line jsx-a11y/anchor-is-valid, jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events
      <a onClick={item.onClick}>{item.display}</a>
    );
  }

  static renderMenuItem(item) {
    if (item.to) {
      return <Link to={item.to}>{item.display}</Link>;
    }
    return DropdownMenu.renderAnchorMenuItem(item);
  }

  renderMenu() {
    const { items } = this.props;
    return (
      <Menu>
        {items.map(item => (
          <Menu.Item key={item.display}>
            {DropdownMenu.renderMenuItem(item)}
          </Menu.Item>
        ))}
      </Menu>
    );
  }

  render() {
    const { title, titleClassName } = this.props;
    return (
      <div>
        <Dropdown overlay={this.renderMenu()}>
          <span className={titleClassName}>
            {title} <Icon type="down" />
          </span>
        </Dropdown>
      </div>
    );
  }
}

DropdownMenu.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      onClick: PropTypes.func,
      href: PropTypes.string,
      to: PropTypes.string,
      display: PropTypes.string.isRequired,
      target: PropTypes.string,
    })
  ).isRequired,
  title: PropTypes.node.isRequired,
  titleClassName: PropTypes.string,
};

DropdownMenu.defaultProps = {
  titleClassName: undefined,
};

export default DropdownMenu;
