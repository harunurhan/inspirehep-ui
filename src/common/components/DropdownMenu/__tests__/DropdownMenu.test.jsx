import React from 'react';
import { shallow } from 'enzyme';

import DropdownMenu from '../DropdownMenu';

describe('DropdownMenu', () => {
  it('renders links without titleClassName', () => {
    const items = [
      {
        display: 'Router Link',
        to: '/router-link',
      },
      {
        display: 'Non Router Link',
        href: '/non-router-link',
      },
      {
        display: 'Non Router Link Button',
        onClick: () => null,
      },
      {
        display: 'External Link',
        href: '//external.link',
        target: '_blank',
      },
    ];
    const wrapper = shallow(<DropdownMenu items={items} title="Test" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders links with titleClassName', () => {
    const items = [
      {
        display: 'Router Link',
        to: '/router-link',
      },
      {
        display: 'Non Router Link',
        href: '/non-router-link',
      },
      {
        display: 'Non Router Link Button',
        onClick: () => null,
      },
      {
        display: 'External Link',
        href: '//external.link',
        target: '_blank',
      },
    ];
    const wrapper = shallow(
      <DropdownMenu items={items} title="Test" titleClassName="test-title" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
