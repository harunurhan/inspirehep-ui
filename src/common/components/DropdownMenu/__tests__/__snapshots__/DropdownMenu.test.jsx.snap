// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu renders links with titleClassName 1`] = `
<div>
  <Dropdown
    mouseEnterDelay={0.15}
    mouseLeaveDelay={0.1}
    overlay={
      <Menu
        className=""
        prefixCls="ant-menu"
        theme="light"
      >
        <MenuItem>
          <Link
            replace={false}
            to="/router-link"
          >
            Router Link
          </Link>
        </MenuItem>
        <MenuItem>
          <a
            href="/non-router-link"
            target={undefined}
          >
            Non Router Link
          </a>
        </MenuItem>
        <MenuItem>
          <a
            onClick={[Function]}
          >
            Non Router Link Button
          </a>
        </MenuItem>
        <MenuItem>
          <a
            href="//external.link"
            target="_blank"
          >
            External Link
          </a>
        </MenuItem>
      </Menu>
    }
    placement="bottomLeft"
    prefixCls="ant-dropdown"
  >
    <span
      className="test-title"
    >
      Test
       
      <Icon
        type="down"
      />
    </span>
  </Dropdown>
</div>
`;

exports[`DropdownMenu renders links without titleClassName 1`] = `
<div>
  <Dropdown
    mouseEnterDelay={0.15}
    mouseLeaveDelay={0.1}
    overlay={
      <Menu
        className=""
        prefixCls="ant-menu"
        theme="light"
      >
        <MenuItem>
          <Link
            replace={false}
            to="/router-link"
          >
            Router Link
          </Link>
        </MenuItem>
        <MenuItem>
          <a
            href="/non-router-link"
            target={undefined}
          >
            Non Router Link
          </a>
        </MenuItem>
        <MenuItem>
          <a
            onClick={[Function]}
          >
            Non Router Link Button
          </a>
        </MenuItem>
        <MenuItem>
          <a
            href="//external.link"
            target="_blank"
          >
            External Link
          </a>
        </MenuItem>
      </Menu>
    }
    placement="bottomLeft"
    prefixCls="ant-dropdown"
  >
    <span>
      Test
       
      <Icon
        type="down"
      />
    </span>
  </Dropdown>
</div>
`;
